pkg/types/azure/validation/platform.go

```diff
@@ -41,3 +45,14 @@ func ValidatePlatform(p *azure.Platform, publish types.PublishingStrategy, fldPa
 	}
 	return allErrs
 }
+
+// ValidateClusterName confirms that the provided cluster name matches Azure naming requirements.
+func ValidateClusterName(clusterName string) error {
+	azureResourceFmt := `[a-z][a-z0-9-]{1,61}[a-z0-9]`
```

Comment:
this is pretty much the same as the GCP one..??

Reply:
Pretty much. Only the regex is different.

Reply:
Is this change good to be merged or am I missing something?

Reply:
as @abhinavdahiya mentioned, it looks like only the grouping is different. @rna-afk if you can confirm that, perhaps we should consolidate this into a single validation.

Reply:
Based on the regex, the difference is that Azure accepts a minimum of 3 characters as the cluster name while the GCP one accepts a minimum of 2. It is a small difference but names like "is" will be accepted in GCP but will not be accepted in the Azure. They are different only in this scenario and I think keeping them separate to avoid this scenario is reason enough.